### PR TITLE
Codespeak language and manual

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
add: Syndicate agents can purchase a "codespeak manual", that teaches
them a language that sounds like a series of codewords. You can also hit
other people with the manual to teach them. One use per manual.
add: Nuclear operatives have access to a deluxe manual that is more
expensive but has unlimited uses.
add: Syndicate AIs know Codespeak for free.
/:cl:

Allows for an additional method of communication and/or security for syndicates. Use of codespeak over the syndicate channel will add additional protection if the channel is compromised, operatives may wish to invest in a deluxe codebook. Also allows for assistants to randomly scream codewords on the radio to annoy people.

- Syndicate codespeak only displays the language icon if you KNOW the
language, so you cannot tell the difference between someone just
shouting random codewords on the radio, and someone actually using the
language.
- Modifies generate_code_phrase to optionally return a list of words,
rather than a preformatted string.
- Codespeak manuals turn into random books after being used.

### CODE REVIEW

**Comment by XDTM** [comment]
I would've preferred to have "Noun verbs noun" or polyspeak as the pattern, but this is still great

**Comment by coiax** [comment]
@XDTM If you like, after this PR, you could change how the scrambling works. I don't particularly mind.

**Comment by BeeSting12** [comment]
so this will appear to everyone else as random codewords... personally i'd search anyone shouting random codewords on comms especially if it's multiple times

**Comment by kevinz000** [comment]
@BeeSting12 that's cause ur shitcurity.

**Comment by XDTM** [comment]
>so this will appear to everyone else as random codewords... personally i'd search anyone shouting random codewords on comms especially if it's multiple times

This is why we need metabreakers, no sportmanship, only valids

**Comment by coiax** [comment]
Also, this will be affected by #26682 until I fix that.

**Comment by FantasticFwoosh** [comment]
Could the syndicate agents on lavaland (the ruin) have pre-loaded code-speak into them? It would be useful with communicating with ops and other code speak syndicate traitors via the communicator role.

**Comment by BeeSting12** [comment]
Screaming random gibberish on comms makes you suspicious... At very least I'd be looking to confiscate your headset for the radio spam, it would get annoying after a while. I can see how this would be useful to nuke ops if their channel were compromise though.

**Comment by BeeSting12** [comment]
would librarians know codespeak?

**Comment by coiax** [comment]
Currently Librarians would know Codespeak.

@FantasticFwoosh yes, that would be easy to do.

**Comment by ExcessiveUseOfCobblestone** [comment]
There is a metabreaker though. Instead of roundstart saying "VIVA" wait a second then start typing in a bunch of codewords.

**Comment by coiax** [comment]
Clearly we need a "syndicate codebook" toy in the arcade machines that just spouts random codewords.

**Comment by XDTM** [comment]
I read that as syndicate cookbook and now i want forbidden recipes for tator chefs

**Comment by coiax** [comment]
Sounds neat. Not sure what they'd be able to cook though.

**Comment by Core0verload** [comment]
I don't like it. If you want a secure channel, use syndicate headset. If you want to detect other traitors, just find a way to disguise real sender and send out code words.

**Comment by KorPhaeron** [comment]
It's already obnoxious enough that people are screaming codewords on the radio without there being a whole language about it

**Comment by Armhulen** [comment]
Well actually WITH SOMEONE'S CHANGE there's now an icon there instead of colors so people would be able to tell if you're speaking codespeak or just talking out of your ass

**Comment by Armhulen** [comment]
so we should totally revert that icon change haHA miright guys

**Comment by coiax** [comment]
@Armhulen if you don't understand syndicate codespeak, you don't get an icon. Meaning it looks the same as Galactic Common.

**Comment by Anonmare** [comment]
So how does codespeak look like if it's used in a typical example? Let's say I said
"Ai is subverted, cap has the disk. Will tell AI to cut comms when ready."

How would it look in codespeak?

**Comment by Niknakflak** [comment]
Seems... kinda.... odd and dumb?

**Comment by optimumtact** [comment]
Please add a snippet to your PR OP that explains the reasoning behind this change

**Comment by coiax** [comment]
@optimumtact done.

File: code/modules/language/codespeak.dm [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+		return
+	playsound(loc, "punch", 25, 1, -1)
+
+	if(M.stat == DEAD)
+		M.visible_message("<span class='danger'>[user] smacks [M]'s lifeless corpse with [src].</span>", "<span class='userdanger'>[user] smacks your lifeless corpse with [src].</span>", "<span class='italics'>You hear smacking.</span>")
+	else if(M.has_language(/datum/language/common/codespeak))
+		M.visible_message("<span class='danger'>[user] beats [M] over the head with [src]!</span>", "<span class='userdanger'>[user] beats you over the head with [src]!</span>", "<span class='italics'>You hear smacking.</span>")
+	else
+		M.visible_message("<span class='notice'>[user] teaches [M] by beating them over the head with [src]!</span>", "<span class='boldnotice'>As [user] hits you with [src], codewords and responses flow through your mind.</span>", "<span class='italics'>You hear smacking.</span>")
+		M.grant_language(/datum/language/common/codespeak)
```
Comment (ChangelingRain) [inline code/modules/language/codespeak.dm]: smacking somebody doesn't use a charge, hmm...

**Comment by coiax** [comment]
@ChangelingRain fixed.

**Comment by optimumtact** [comment]
lets wait until we see how the lizard thing goes.

**Comment by coiax** [comment]
This was opened before Draconic, and one of the first things me and @XDTM talked about as a possibility of the extension of the language features.

**Comment by optimumtact** [comment]
I really don't care

**Comment by ChangelingRain** [comment]
BUT I DO, ORANGES, I DO

**Comment by Militaires** [comment]
# CITRUS WAR 3: WORLD OF PAIN

**Comment by kevinz000** [comment]
STOP

**Comment by KorPhaeron** [comment]
I know this wont stop joan and oranges but we dont need peanut gallery on top of that

**Comment by coiax** [comment]
@Cheridan @KorPhaeron @optimumtact @ChangelingRain 
Can we resolve this stalemate? It's not a reactive lizard language PR, it doesn't really conflict.

**Comment by GunHog** [comment]
Is there a way to teach this to a Syndicate AI?

**Comment by coiax** [comment]
@GunHog well, currently you can beat the AI with a syndicate codebook. Could also give it to a syndicate AI for free. Thoughts?

**Comment by GunHog** [comment]
Well, it is logical since they are Syndicate and they could easily program it into their AI agents. A low cost or free unlock for them is good as they do not have uplinks.

**Comment by Armhulen** [comment]
Malf ais are not syndicates, that is lore unfriendly

**Comment by GunHog** [comment]
> Malf ais are not syndicates, that is lore unfriendly

Malf and Traitor AI are now the same thing, as proper Malf was removed. Syndicate AIs appear in Traitor modes. Instead of "Malfunction", it is technically a combat software given to the AI by the Syndicate. I still erroneously the CPU time "Malf Points".

Because of this, it is entirely lore-friendly!

**Comment by optimumtact** [comment]
Where did you pull that lore from

**Comment by Anonmare** [comment]
>Caring about lore in horizontal sprite simulator

**Comment by Iamgoofball** [comment]
# stop involving lore in game balance

**Comment by Iamgoofball** [comment]
# stop involving lore in game balance

**Comment by Anonmare** [comment]
Make more, not lore okay?

**Comment by KorPhaeron** [comment]
Malf AI gets a free syndicate radio so it is probably syndicate

File: code/modules/language/codespeak.dm [hunk 2]
```diff
@@ -19,6 +19,10 @@
 
 	. = capitalize(.)
 
+	var/input_ending = copytext(input, length(input))
+	if(input_ending in list("!","?","."))
```
Comment (RemieRichards) [inline code/modules/language/codespeak.dm]: sad.

Comment (Cyberboss) [inline code/modules/language/codespeak.dm]: static

**Comment by optimumtact** [comment]
@tgstation/commit-access 24 hours 2 merge or I'm closing this

**Comment by coiax** [comment]
Yeah, better move quickly, or the fruit will continue his exciting non-vendetta against everything I do.

**Comment by optimumtact** [comment]
I'm not going to leave things on the tracker forever, @ChangelingRain obviously wanted this change but not enough to do the review and merge

Since I don't have any requirement to review and merge your pr, I"ll just close it and you can sort out with @ChangelingRain the details of getting it reviewed and merged at your leisure.

**Comment by lzimann** [comment]
I don't like this

**Comment by Dorsisdwarf** [comment]
I'm against this change for as long as the librarian can speak this 2-tc language, negating half the point.

**Comment by XDTM** [comment]
It's a 2 tc language, of course it's not everything-proof. Unlike comm keys they can't be stolen, so they instead have the librarian weakness.

**Comment by coiax** [comment]
Also, if I remove Codespeak from the Librarian, you just know that @ChangelingRain will use that as an excuse to remove Ratvarian from them as well.

**Comment by coiax** [comment]
@KorPhaeron Yes or no?

**Comment by coiax** [comment]
Because @ChangelingRain wants to ask you first.

**Comment by optimumtact** [comment]
@KorPhaeron has no strong opinion on this PR

@ChangelingRain up to you now

**Pull request merged**